pkg/apis/configuration/validation/validation.go

```diff
@@ -857,11 +898,38 @@ func validateVirtualServerRouteHost(host string, virtualServerHost string, field
 	return allErrs
 }
 
+var regexOrExactMatchPrefixes = []string{"~", "="}
+
+func isRegexOrExactMatch(path string) bool {
+	for _, p := range regexOrExactMatchPrefixes {
+		if strings.HasPrefix(path, p) {
+			return true
+		}
+	}
+
+	return false
+}
+
 func validateVirtualServerRouteSubroutes(routes []v1alpha1.Route, fieldPath *field.Path, upstreamNames sets.String, pathPrefix string) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	allPaths := sets.String{}
 
+	if isRegexOrExactMatch(pathPrefix) {
```

Comment:
it seems that this could be simplified. additionally, we can return in the if block  - no need to go into the loop:
```go
	if isRegexOrExactMatch(pathPrefix) {
		if len(routes) != 1 {
			return append(allErrs, field.Invalid(fieldPath, "subroutes", "must have only one subroute if regex match or exact match are being used"))
		}

		if routes[0].Path != pathPrefix {
			return append(allErrs, field.Invalid(fieldPath.Index(0).Child("path"), routes[0].Path, "must have the same path as the referenced VirtualServer route path"))
		}

		isRouteFieldForbidden := true
		return validateRoute(routes[0], fieldPath.Index(0).Child("path"), upstreamNames, isRouteFieldForbidden)
	}
```